libnavigation-base/src/main/java/com/mapbox/navigation/base/options/PredictiveCacheOptions.kt

```diff
@@ -71,17 +82,42 @@ class PredictiveCacheOptions private constructor(
 
         /**
          * Predictive cache Maps related options.
+         * @deprecated use predictiveCacheMapsOptionsList instead to provide different
+         * [PredictiveCacheLocationOptions]s for different zoom level ranges.
          */
+        @Deprecated(
+            "Use predictiveCacheMapsOptionsList",
+            ReplaceWith("predictiveCacheMapsOptionsList(listOf(predictiveCacheMapsOptions))")
+        )
         fun predictiveCacheMapsOptions(
             predictiveCacheMapsOptions: PredictiveCacheMapsOptions
-        ): Builder = apply { this.predictiveCacheMapsOptions = predictiveCacheMapsOptions }
+        ): Builder = predictiveCacheMapsOptionsList(listOf(predictiveCacheMapsOptions))
+
+        /**
+         * List of predictive cache Maps related options ([PredictiveCacheMapsOptions]).
+         * Use this instead of [predictiveCacheMapsOptions] so that you can specify different
+         * [PredictiveCacheLocationOptions]s for different zoom level ranges.
+         *
+         * @throws IllegalArgumentException if [predictiveCacheMapsOptionsList] is empty.
+         */
+        @Throws(IllegalArgumentException::class)
+        fun predictiveCacheMapsOptionsList(
+            predictiveCacheMapsOptionsList: List<PredictiveCacheMapsOptions>
+        ): Builder = apply {
+            if (predictiveCacheMapsOptionsList.isEmpty()) {
+                throw IllegalArgumentException("predictiveCacheMapsOptionsList must not be empty")
+            }
+            this.predictiveCacheMapsOptionsList = predictiveCacheMapsOptionsList.toList()
```

Comment:
```suggestion
            this.predictiveCacheMapsOptionsList = predictiveCacheMapsOptionsList
```

Reply:
I want the copy. So that if a user modifies the list later, it's not reflected in our list.
It's just the approach I try to follow with user-provided collections.